### ISSUE TITLE
Plug in gas estimation in cairo runner

### ISSUE THREAD

**Issue by Arcticae** [software-mansion/protostar#1303]

**Comment by MaksymilianDemitraszek** [comment]
As right now default test runtime won't be on-chain we may reconsider how do we want to report gas used by tests. Maybe we want to collect and sum up cost of all of the on-chain operations executed? Or we want to keep reporting explicit and add `estimate` cheatcode?

**Comment by Arcticae** [comment]
@MaksymilianDemitraszek actually i think we should reconsider since all functions will be consuming a set amount of gas in cairo 1.0, so i don't know if reporting this would be either much easier or just unnecessary.

**Comment by MaksymilianDemitraszek** [comment]
I think we still need it, but I would keep it out of MVP

**Comment by Arcticae** [comment]
I'd say it's blocked by [cairo 1.0 integration](https://github.com/software-mansion/protostar/milestone/31), since we'll need execution resources (gas) returned by the VM 

cc @MaksymilianDemitraszek

**Comment by MaksymilianDemitraszek** [comment]
Yup, I would keep it out of protostar 1.0 release

**Comment by MaksymilianDemitraszek** [comment]
Though i don't think it is blocked by cairo 1 integration, resources estimation is a pure-starknet thing. We may restrain until starknet 0.11 though

**Comment by Arcticae** [comment]
So still - it's blocked. If starknet 0.11 exposes this functionality, we might consider including it, if we have spare time.

**Comment by MaksymilianDemitraszek** [comment]
> So still - it's blocked.
Blocked means we cannot work on that because of technical coinstraints, better to be precise

**Comment by Arcticae** [comment]
@MaksymilianDemitraszek isn't gas consumption something that's supposed to be on the cairo-vm side (from what we've heard at the conf?)

**Comment by MaksymilianDemitraszek** [comment]
Yes it does, calculating the total gas consumption is roughly equivaltent to executing
